### PR TITLE
Revert "Revert "[core] Fix placement groups scheduling when no resources are specified""

### PR DESCRIPTION
**Pull request by vitsai**
Reverts ray-project/ray#40506

Fix originally in https://github.com/ray-project/ray/pull/40507 is also merged into here.

### CODE REVIEW

File: .buildkite/ml.rayci.yml [hunk 1]
```diff
@@ -11,6 +11,27 @@ steps:
...
+    tags: tune
+    instance_type: large
+    commands:
+      - bazel run //ci/ray_ci:test_in_docker -- //python/ray/tune/... ml 
+        --parallelism-per-worker 3
+        --except-tags soft_imports,gpu_only,rllib,multinode
+        --test-env=AIR_VERBOSITY=1
+    depends_on: mlbuild
+    job_env: forge
+
```
Comment (jjyao) [inline .buildkite/ml.rayci.yml:34]: ?

File: dashboard/state_aggregator.py [hunk 2]
```diff
@@ -408,17 +407,14 @@ async def list_tasks(self, *, option: ListApiOptions) -> ListApiResponse:
             protobuf_to_task_state_dict(message) for message in reply.events_by_task
         ]
 
-        # Num pre-truncation is the number of tasks returned from
-        # source + num filtered on source
-        num_after_truncation = len(result) + reply.num_filtered_on_gcs
-        num_total = reply.num_total_stored + reply.num_status_task_events_dropped
+        num_after_truncation = len(result)
```
Comment (rkooo567) [inline dashboard/state_aggregator.py:410]: why are these changed?

File: .buildkite/core.rayci.yml [hunk 3]
```diff
@@ -68,7 +68,6 @@ steps:
         --test-env=TEST_EXTERNAL_REDIS=1
         --test-env=EXPECTED_PYTHON_VERSION={{matrix}}
         --only-tags minimal
-        --except-tags no_redis
         --skip-ray-installation
```
Comment (rkooo567) [inline .buildkite/core.rayci.yml:71]: related?

**Review by rkooo567 — COMMENTED** [review]
what was the issue from the previous failure?

**Comment by vitsai** [comment]
Ah sorry, should have closed this. Look at this one instead: https://github.com/ray-project/ray/pull/40587
